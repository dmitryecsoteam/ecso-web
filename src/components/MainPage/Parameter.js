import React from 'react';
import InputRange from 'react-input-range';

export default (props) => {
    return (
        <div>
            <label>{props.parameter}</label>
            <InputRange
                maxValue={5}
                minValue={0}
                value={props.value}
                onChange={(e) => props.onChange(e, props.parameter)}
                disabled={props.disabled}
            />
        </div>
)};