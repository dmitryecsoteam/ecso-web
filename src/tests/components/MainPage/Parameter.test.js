import React from 'react';
import { shallow } from 'enzyme';
import Parameter from '../../../components/MainPage/Parameter';


const testProps = {
    parameter: "Test parameter",
    value: 2,
    onChange: () => {},
    disabled: false
};

test('should render Parameter component with default props', () => {
    const wrapper = shallow(<Parameter {...testProps} />);
    expect(wrapper).toMatchSnapshot();
});

test('should pass new value in onChange function on direct input change', () => {
    const onChange = jest.fn(); 
    const wrapper = shallow(<Parameter {...{...testProps, onChange}} />);
    wrapper.find('input').simulate('change', { target: { value: 4 }});
    expect(onChange).toBeCalledWith(4, "Test parameter");
});

test('should pass new value in onChange function on label click', () => {
    const onChange = jest.fn(); 
    const wrapper = shallow(<Parameter {...{...testProps, onChange}} />);
    wrapper.find('.slider__label').at(0).simulate('click', { target: { innerHTML: "0" }});
    expect(onChange).toBeCalledWith(0, "Test parameter");
});