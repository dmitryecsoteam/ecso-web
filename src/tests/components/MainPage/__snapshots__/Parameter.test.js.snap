// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Parameter component with default props 1`] = `
<div
  className="parameter-container"
>
  <label
    className="parameter__label"
  >
    Test parameter
  </label>
  <div
    className="slider-container"
  >
    <div
      className="slider__labels-container"
    >
      <div
        className="slider__label slider__label--active slider__label--first"
        key="0"
        onClick={[Function]}
      >
        0
      </div>
      <div
        className="slider__label slider__label--active"
        key="1"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="slider__label slider__label--active slider__label--selected"
        key="2"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="slider__label"
        key="3"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="slider__label"
        key="4"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="slider__label slider__label--last"
        key="5"
        onClick={[Function]}
      >
        5
      </div>
    </div>
    <div
      className="slider__range-contaner"
    >
      <input
        className="slider__range"
        disabled={false}
        max={5}
        min={0}
        onChange={[Function]}
        steps={1}
        style={
          Object {
            "background": "linear-gradient(to right, optionChoose 40%, #b2b2b2 40%)",
          }
        }
        type="range"
        value={2}
      />
    </div>
  </div>
</div>
`;
